pkg/ha/ha_super_stream_publisher_test.go

```diff
@@ -0,0 +1,144 @@
+package ha
+
+import (
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/amqp"
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/message"
+	test_helper "github.com/rabbitmq/rabbitmq-stream-go-client/pkg/test-helper"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream"
+)
+
+var _ = Describe("Reliable Super Stream Producer", func() {
+
+	var (
+		envForSuperStreamProducer    *Environment
+		streamForSuperStreamProducer string
+	)
+	BeforeEach(func() {
+		testEnv, err := NewEnvironment(nil)
+		envForSuperStreamProducer = testEnv
+		Expect(err).NotTo(HaveOccurred())
+		streamForSuperStreamProducer = fmt.Sprintf("super_stream_producer_%s", uuid.New().String())
+		err = envForSuperStreamProducer.DeclareSuperStream(streamForSuperStreamProducer, NewPartitionsOptions(3))
+		Expect(err).NotTo(HaveOccurred())
+	})
+	AfterEach(func() {
+		Expect(envForSuperStreamProducer.DeleteSuperStream(streamForSuperStreamProducer)).NotTo(HaveOccurred())
+	})
+
+	It("Validate mandatory fields", func() {
+		_, err := NewReliableSuperStreamProducer(envForSuperStreamProducer,
+			streamForSuperStreamProducer, nil, nil)
+		Expect(err).To(HaveOccurred())
+		_, err = NewReliableSuperStreamProducer(envForSuperStreamProducer,
+			streamForSuperStreamProducer, &SuperStreamProducerOptions{}, nil)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Create/Confirm and close a Reliable Super Stream Producer", func() {
+
+		signal := make(chan struct{})
+		var confirmed int32
+		superProducer, err := NewReliableSuperStreamProducer(envForSuperStreamProducer,
+			streamForSuperStreamProducer, &SuperStreamProducerOptions{
+				RoutingStrategy: NewHashRoutingStrategy(func(message message.StreamMessage) string {
+					return message.GetApplicationProperties()["routingKey"].(string)
+				}),
+			}, func(messageConfirm []*PartitionPublishConfirm) {
+				for _, confirm := range messageConfirm {
+					for i, status := range confirm.ConfirmationStatus {
+						Expect(status.IsConfirmed()).To(BeTrue(), fmt.Sprintf("Message %d not confirmed", i))
+						if atomic.AddInt32(&confirmed, 1) == 200 {
+							signal <- struct{}{}
+						}
+					}
+				}
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		for i := 0; i < 200; i++ {
+			msg := amqp.NewMessage(make([]byte, 0))
+			msg.ApplicationProperties = map[string]any{"routingKey": fmt.Sprintf("hello%d", i)}
+			err = superProducer.Send(msg)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		<-signal
```

Comment:
```suggestion
		select {
		case <-signal:
		case <-time.After(5 * time.Second):
			Fail("Timeout waiting messages")
		}
```
To avoid pending failing jobs.